test/e2e/features_test.go

```diff
@@ -0,0 +1,334 @@
+HNNCu1tu56NTMFEwHQYDVR0OBBYEFEG94gMq4SvGtTs48Nw5BzVnPK69MB8GA1Ud
+IwQYMBaAFEG94gMq4SvGtTs48Nw5BzVnPK69MA8GA1UdEwEB/wQFMAMBAf8wCgYI
+KoZIzj0EAwIDaQAwZgIxAPRJkWfSdIQMr2R77RgCicR+adD/mMxZra2SoL7qSMyq
+3iXLIXauNP9ar3tt1uZE8wIxAM4C6G4uoQ0dydhcgQVhlgB6GaqO18AEDYPzQjir
+dV2Bs8EBkYBx87PmZ+e/S7g9Ug==
+-----END CERTIFICATE-----`,
+			Key: `-----BEGIN PRIVATE KEY-----
+MIG2AgEAMBAGByqGSM49AgEGBSuBBAAiBIGeMIGbAgEBBDBVtvjDBFke/k2Skezl
+h63g1q5IHCQM7wr1T43m5ACKZQt0ZDE1jfm1BYKk1omNpeChZANiAARwk2G6qdz9
+r+Wg7oZDvta1TQInW9FHJwnbcqGcdPjqs9+QLqvxjWi72UWLMlukh1RmkVoT4d40
+PMxZ3NvEwhsJgDJ82D7OUR2G7wZtgUj/WFj14XOofpZJmhzTQrtbbuc=
+-----END PRIVATE KEY-----`,
+		},
+	}
+)
+
+// webhookKINDConfig is a KIND configuration used for TestWebhookUpdate. KIND, when running in GitHub Actions, is
+// a bit wonky with handling Secret updates, and they do not propagate to container filesystems in a reasonable
+// amount of time (>10m) when running this in the complete test suite, even though the actual sync frequency/update
+// propagation should be 1m by default. These changes force Secret updates to go directly to the API server and
+// update containers much more often. The latter causes significant performance degradation elsewhere, and Pods take
+// much longer to start, but once they do Secret updates show up more quickly, enough for the test to complete in time.
+const webhookKINDConfig = `kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  kubeadmConfigPatches:
+  - |
+    kind: KubeletConfiguration
+    configMapAndSecretChangeDetectionStrategy: Get
+    syncFrequency: 3s
+`
+
+// TestWebhookUpdate checks that the webhook updates the certificate indicated by --admission-webhook-cert-file when
+// the mounted Secret updates. This requires E2E because we can't mount Secrets with the locally-run integration
+// test controller instance.
+func TestWebhookUpdate(t *testing.T) {
+	t.Log("configuring all-in-one-dbless.yaml manifest test")
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	t.Log("building test cluster and environment")
+	configFile, err := os.CreateTemp(os.TempDir(), "webhook-kind-config-")
+	require.NoError(t, err)
+	defer os.Remove(configFile.Name())
+	defer configFile.Close()
+	written, err := configFile.Write([]byte(webhookKINDConfig))
+	require.NoError(t, err)
+	require.Equal(t, len(webhookKINDConfig), written)
+
+	clusterBuilder := kind.NewBuilder()
+	clusterBuilder.WithConfig(configFile.Name())
+	if clusterVersionStr != "" {
+		clusterVersion, err := semver.ParseTolerant(clusterVersionStr)
+		require.NoError(t, err)
+		clusterBuilder.WithClusterVersion(clusterVersion)
+	}
+	cluster, err := clusterBuilder.Build(ctx)
+	require.NoError(t, err)
+	addons := []clusters.Addon{}
+	addons = append(addons, metallb.New())
+	if b, err := loadimage.NewBuilder().WithImage(imageLoad); err == nil {
+		addons = append(addons, b.Build())
+	}
```

Comment:
The way we're handling the error here seems a bit off, it seems we should check whether imageLoad is populated, if so run `WithImage` and then require no error.

Reply:
Follows the existing pattern from elsewhere, though I agree it's a bit wonky, so https://github.com/Kong/kubernetes-ingress-controller/issues/2287 for later.